discord_bots/cogs/admin.py

```diff
@@ -0,0 +1,692 @@
+            if len(players) == 0:
+                session.add(
+                    Player(
+                        id=member.id,
+                        name=member.name,
+                        is_banned=True,
+                    )
+                )
+                await interaction.response.send_message(
+                    embed=Embed(
+                        description=f"{escape_markdown(member.name)} banned",
+                        colour=Colour.green(),
+                    )
+                )
+                session.commit()
+            else:
+                player = players[0]
+                if player.is_banned:
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description=f"{escape_markdown(player.name)} is already banned",
+                            colour=Colour.red(),
+                        )
+                    )
+                else:
+                    player.is_banned = True
+                    session.commit()
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description=f"{escape_markdown(player.name)} banned",
+                            colour=Colour.green(),
+                        )
+                    )
+
+    @group.command(
+        name="configure", description="Initially configure the bot for this server"
+    )
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    async def configure(self, interaction: Interaction):
+        if interaction.guild:
+            session: SQLAlchemySession
+            with Session() as session:
+                guild = (
+                    session.query(DiscordGuild)
+                    .filter(DiscordGuild.discord_id == interaction.guild.id)
+                    .first()
+                )
+                if guild:
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description="Server already configured",
+                            colour=Colour.red(),
+                        ),
+                        ephemeral=True,
+                    )
+                else:
+                    guild = DiscordGuild(interaction.guild.id, interaction.guild.name)
+                    session.add(guild)
+                    session.commit()
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description="Server configured successfully!",
+                            colour=Colour.green(),
+                        ),
+                        ephemeral=True,
+                    )
+        else:
+            await interaction.response.send_message(
+                embed=Embed(
+                    description="Command must be run from within a guild",
```

Comment:
the `app_commands.guild_only` decorator can get rid of this check, if you want to simplify it a bit. Just `assert guild` at the top if you do that

Reply:
Added `@app_commands.guild_only()`, and simplified function logic to remove the `if interaction.guild`